components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java

```diff
@@ -2254,4 +2259,113 @@ public static String getAgentIdentityUserstoreName() {
         }
         return userStoreName;
     }
+
+    /**
+     * Check whether the JWT exceeds the allowed depth.
+     *
+     * @param jwt JWT string to check.
+     * @throws  ParseException If the JWT exceeds the allowed depth or if an error occurs during parsing.
+     */
+    public static void validateJWTDepth(String jwt) throws ParseException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Checking JWT depth validation");
+        }
+
+        if (StringUtils.isBlank(jwt)) {
+            if (log.isDebugEnabled()) {
+                log.debug("JWT is blank, skipping depth validation");
+            }
+            throw new ParseException("Error validating JWT depth. JWT is blank.", 0);
+        }
+
+        // Extract and decode JWT payload.
+        String[] parts = jwt.split("\\.");
+        if (parts.length < 2) {
+            log.warn("Invalid JWT format");
```

Comment:
Do we need to warn and throw error both here?

Reply:
Removed with https://github.com/wso2/carbon-identity-framework/pull/7368/commits/8ef793d35b1561c4c47e7ec4fed3afe142534cf4